scripts/safe_template_checker.py

```diff
@@ -0,0 +1,498 @@
+#!/usr/bin/env python
+"""
+a tool to check if templates are safe
+"""
+from enum import Enum
+import os
+import re
+import sys
+
+_skip_dirs = (
+    '/node_modules',
+    '/vendor',
+    '/spec',
+    '/.pycharm_helpers',
+    '/test_root',
+    '/reports/diff_quality',
+    '/common/static/xmodule/modules',
+)
+_skip_mako_dirs = _skip_dirs
+_skip_underscore_dirs = _skip_dirs + ('/test',)
+
+
+def _is_skip_dir(skip_dirs, directory):
+    for skip_dir in skip_dirs:
+        if (directory.find(skip_dir + '/') >= 0) or directory.endswith(skip_dir):
+            return True
+    return False
+
+
+def _load_file(self, file_full_path):
+    input_file = open(file_full_path, 'r')
+    try:
+        file_contents = input_file.read()
+    finally:
+        input_file.close()
+
+    if not file_contents:
+        return False
```

Comment:
Agreed. When we last saw this together I was guessing this was the result of an improper refactor.